flycheck.el

```diff
@@ -7429,6 +7429,17 @@ This option requires Go 1.6 or newer."
                  (const :tag "Check for shadowed variables" t)
                  (const :tag "Strictly check for shadowed variables" strict)))
 
+(flycheck-def-option-var flycheck-go-megacheck-disabled-checkers nil go-megacheck
+  "A list of checkers to disable when running `megacheck'.
+
+The value of this variable is a list of strings, where each
+string is a checker to be disabled. Valid checkers are `simple',
+`staticcheck' and `unused'. When nil, all checkers will be
+enabled. "
+  :type '(choice (repeat :tag "Disabled checkers"
+                         (string :tag "Checker name")))
```

Comment:
I think you can use `const` instead of `string` here to not allow arbitrary strings, but only the names of the checkers that make sense.